sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -30,8 +32,14 @@ public abstract class TestEnvironment
 
         private TokenCredential _credential;
         private TestRecording _recording;
+        private readonly string _serviceName;
 
-        private readonly Dictionary<string, string> _environmentFile = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        private Dictionary<string, string> _environmentFile;
+        private readonly string _serviceSdkDirectory;
+
+        private bool _bootstrappingAttempted;
+        private readonly SemaphoreSlim _bootstrappingSemaphore = new(1, 1);
```

Comment:
We create an instance of TestEnvironment per test class. This semaphore might not be enough